lib/datadog/tracing/contrib/status_range_env_parser.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Tracing
+    module Contrib
+      # Parsing status range from environment variable
+      class StatusRangeEnvParser
+        class << self
+          def call(value)
+            [].tap do |array|
+              value.split(',').each do |e|
+                next unless e
+
+                v = e.split('-')
+
+                case v.length
+                when 0 then next
+                when 1 then array << Integer(v.first)
+                when 2 then array << (Integer(v.first)..Integer(v.last))
+                else
+                  Datadog.logger.debug(
+                    "Invalid error_status_codes configuration: Unable to parse #{value}, containing #{v}."
+                  )
```

Comment:
Somehow I feel like this logging should happen in `o.env_parser`.

Also, I think we might be inconsistent and in some places env parse fails / conf type checks raise and in others they log but IIRC we mostly raise? I might be wrong on that one but I'd rather be consistent.

If we chose to always log, I feel like we should raise `EnvValueError` / `ParseError` / `TypeError` / whatever exceptions in such pieces of code, and then the caller (the config system) of `o.env_parser` / `o.setter` / ... should handle that kind of exception uniformly and log.